src/Twilio/Http/SystemNetHttpClient.cs

```diff
@@ -91,16 +87,47 @@ private HttpRequestMessage BuildHttpRequest(Request request)
             httpRequest.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
             httpRequest.Headers.AcceptEncoding.Add(new StringWithQualityHeaderValue("utf-8"));
 
-            int lastSpaceIndex = PlatVersion.LastIndexOf(" ");
-            System.Text.StringBuilder PlatVersionSb= new System.Text.StringBuilder(PlatVersion);
-            PlatVersionSb[lastSpaceIndex] = '/';
+            var userAgent = LibraryVersion;
+            if (request.UserAgentExtensions != null)
+            {
+                var userAgentBuilder = new StringBuilder(userAgent);
+                foreach (var extension in request.UserAgentExtensions)
+                {
+                    userAgentBuilder.Append($" {extension}");
+                }
+
+                userAgent = userAgentBuilder.ToString();
+            }
+
+            httpRequest.Headers.TryAddWithoutValidation("User-Agent", userAgent);
+
+            foreach (var header in request.HeaderParams)
+            {
+                httpRequest.Headers.TryAddWithoutValidation(header.Key, header.Value);
+            }
+
+            return httpRequest;
+        }
+
+        private static string BuildLibraryVersion()
+        {
+            var frameworkDescription = RuntimeInformation.FrameworkDescription;
+            var platformVersionSb = new StringBuilder(frameworkDescription);
+            
+            // Twilio runs CI tests on Mono, where the framework description looks like this:
+            // Mono 6.12.0.182 (tarball Tue Jun 14 22:35:00 UTC/2022)
+            // The following code removes the (tarball ...) part from the framework description.
+            if (frameworkDescription.StartsWith("Mono ", StringComparison.Ordinal))
+            {
+                var tarbalDateIndex = frameworkDescription.IndexOf(" (tarball");
+                platformVersionSb.Remove(tarbalDateIndex, frameworkDescription.Length - tarbalDateIndex);
+                frameworkDescription = platformVersionSb.ToString();
+            }
```

Comment:
The CI tests use Mono to run .NET Framework instead of actual Windows .NET Framework.
As a side effect of not hardcoding ".NET Framework 4.5.1+" anymore,, the FrameworkDescription will look like "Mono 6.12.0.182 (tarball Tue Jun 14 22:35:00 UTC/2022)" which doesn't conform with the existing format expection in the tests.
We could change the test to allow the tarball string, or strip it off like I'm doing here.